### PR TITLE
Add intellisense, loading screen, and UI fixes.

### PR DESCRIPTION
**Pull request by csyonghe**
Need to merge slang PR first https://github.com/shader-slang/slang/pull/5419 for additional wasm bindings.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -51,9 +51,60 @@
     <script src="compiler.js"></script>
     <script src="try-slang.js"></script>
     <script src="slang-wasm.js"></script>
+    <script src="language-server.js"></script>
+    <style>
+      #loading-screen {
```
Comment (kaizhangNV) [inline index.html:56]: should we put the style code to css file?

**Review by kaizhangNV — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (csyonghe) [reply index.html:56]: It was originally in the css file, but for some reason my browser couldn't pick them up.
It is likely due to stale browser cache, and we should be able to move them back.

Comment (kaizhangNV) [reply index.html:56]: yes, it must be the cache thing.
I sometimes have to delete the browser cache after I change the code.

**Comment by csyonghe** [comment]
Done.

**Pull request merged**